augur/metrics/commit/commit.py

```diff
@@ -307,6 +312,11 @@ def top_committers(self, repo_group_id, repo_id=None, year=None, threshold=0.5):
 
         results = pd.read_sql(total_commits_SQL, self.database,
                             params={'year': year, 'repo_id': repo_id})
+        if(results.iloc[0]['sum']==None):
+            year -=1
+            results = pd.read_sql(total_commits_SQL, self.database,
+                                params={'year': year, 'repo_id': repo_id})
+
 
     total_commits = int(results.iloc[0]['sum'])
```

Comment:
@pratikmishra356 Hi! Umm, I believe that when we're trying to pull commits for a certain year, augur users would expect that it returns the commits for that very year(even if it has zero commits!).
Ergo, I think that the valid fix would be as follows:

 ```
total_commits = 0 if results.iloc[0]['sum'] is None else int(results.iloc[0]['sum'])
```

Reply:
The repercussions of your current fix would lead to another bug namely, **When commits are zero for a current year, it shows commits for the previous year**.
Just a heads up!

Reply:
@mrsaicharan1 you are right but then in the later code it returns error
```suggestion
    cumsum = 0
    for i, row in results.iterrows():
        cumsum += row['commits']
        if cumsum >= threshold_commits:
            results = results[:i + 1]
            break

    if not repo_id:
        rg_name = results.iloc[0]['repo_group_name']
        results.loc[i+1] = [repo_group_id, rg_name, 'other_contributors',
                            int(total_commits - cumsum)]
```

```suggestion
    rg_name = results.iloc[0]['repo_group_name']
```

Reply:
https://github.com/chaoss/augur/issues/615#issuecomment-605511412

This just shows one error. Please put up the error pertaining to the one mentioned above.

Reply:
![Screen Shot 2020-03-29 at 12 35 13 PM](https://user-images.githubusercontent.com/43684300/77843349-e8fcee00-71b9-11ea-9029-f71062c3d68e.png)
it will show errors for every columns in results ,another way would be to set value "0" for every columns in results if there is no data available.